packages/api/src/hooks/useMobileCarouselWalletsList.ts

```diff
@@ -0,0 +1,39 @@
+import { useState, useEffect } from 'react';
+import useWalletAccountsList from './useWalletAccountsList';
+import useCurrencyConfig from './useCurrencyConfig';
+
+/** A custom hook that gets the list of all wallet accounts for the current user. */
+const useMobileCarouselWalletsList = () => {
+    const { data: walletAccountsList, isLoading } = useWalletAccountsList();
+    const { isLoading: isCurrencyConfigLoading } = useCurrencyConfig();
+
+    const [staleWalletAccountsList, setStaleWalletAccountslist] = useState(walletAccountsList);
+    const [staleIsLoading, setStaleIsLoading] = useState(true);
+
+    useEffect(() => {
+        if (!isLoading && !isCurrencyConfigLoading && walletAccountsList && !staleWalletAccountsList) {
+            setStaleIsLoading(false);
+            setStaleWalletAccountslist(walletAccountsList);
+        }
+
+        // due to quirky nature of underlying hooks chain, need this complex list of conditions
+        if (
```

Comment:
I hope to refactor this condition in subsequent PRs by extracting some logic from the whole chain of hooks into separate functions,